pulser-simulation/pulser_simulation/hamiltonian.py

```diff
@@ -171,13 +272,24 @@ def set_config(self, cfg: NoiseModel) -> None:
                 f"Interaction mode '{self._interaction}' does not support "
                 f"simulation of noise types: {', '.join(not_supported)}."
             )
-        if not hasattr(self, "basis_name"):
+        with_leakage = "leakage" in cfg.noise_types
```

Comment:
What if we do this:
```suggestion
        no_resampling = set(cfg.noise_types).issubset(
            {   # These should go in a CONSTANT
                "dephasing",
                "relaxation",
                "SPAM",
                "depolarizing",
                "eff_noise",
                "leakage",
            })
        # Use leakage to emulate state prep errors when resampling the hamiltonian is not needed
        with_leakage = "leakage" in cfg.noise_types or ("SPAM" in cfg.noise_types and cfg.state_prep_error > 0 and no_resampling)
```

Reply:
Actually, I think it has to be even more involved than this. When the user explicitly adds leakage error, they might define effective noise operators that allow the population of the `x` state to go back to one of the other basis states, which would make `x` unsuitable for emulating the SP errors.
We'll want to distinguish `with_leakage` when it's user selected and when we activate ourselves to use for SP errors.